File: springboot-java/src/main/java/hello/HelloController.java
package hello;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloController {

    private final Foo foo;

    HelloController(Foo foo) {
        this.foo = foo;
    }

    @RequestMapping("/")
    public String index() {
        return "Greetings from Spring Boot!";
    }

    @RequestMapping("/foo")
    public String foo() {
        return foo.getName();
    }

}

File: springboot-kotlin/src/main/kotlin/hello/HelloController.kt
package hello

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.util.concurrent.atomic.AtomicLong

@RestController
class HelloController @Autowired constructor(private val greetingService: GreetingService) {

    val counter = AtomicLong()

    @GetMapping("/person")
    fun greeting(@RequestParam(value = "name", defaultValue = "World") name: String) =
        Greeting(counter.incrementAndGet(), greetingService.greet(name))

    data class Greeting(val id: Long, val content: String)
}
